### PR TITLE
feat:hide subtitles popup.

### PR DESCRIPTION
**Pull request by Aunali321**
Remove this:
image

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.annotation.HideSubtitlesPopupPatchCompatibility
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.fingerprint.HideSubtitlesPopupFingerprint
+
+@Patch
+@DependsOn()
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:17]:
```suggestion
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+@Description("Hides subtitles popup.")
+@HideSubtitlesPopupPatchCompatibility
+@Version("0.0.1")
+class HideSubtitlesPopupPatch : BytecodePatch(
+    listOf(
+        HideSubtitlesPopupFingerprint
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:28]:
```suggestion
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        val subtitlePopupMethod = HideSubtitlesPopupFingerprint.result!!.mutableMethod
+
+        // add a return instruction at the start of the void method
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:31]: Comment duplicates code

```suggestion
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.annotation.HideSubtitlesPopupPatchCompatibility
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.fingerprint.HideSubtitlesPopupFingerprint
+
+@Patch
+@DependsOn()
+@Name("hide-subtitles-popup")
+@Description("Hides subtitles popup.")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:19]:
```suggestion
@Description("Hides the toast notification when toggling subtitles.")
```

**Review by oSumAtrIX — APPROVED** [review]

**Comment by Aunali321** [comment]
This is ready to be merged.

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        val subtitlePopupMethod = HideSubtitlesPopupFingerprint.result!!.mutableMethod
+
+        subtitlePopupMethod.replaceInstructions(
+            0,
+            "return-void"
+        )
+
+        return PatchResultSuccess()
+    }
+
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:36]:
```suggestion
```

**Comment by oSumAtrIX** [comment]
This patch should use the `settings` patch and allow toggling via the settings in favor of UX.

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/annotation/HideSubtitlesPopupPatchCompatibility.kt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package app.revanced.patches.youtube.layout.hidesubtitlespopup.annotation
+
+import app.revanced.patcher.annotation.Compatibility
+import app.revanced.patcher.annotation.Package
+
+@Compatibility([Package("com.google.android.youtube", arrayOf("17.49.37"))])
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/annotation/HideSubtitlesPopupPatchCompatibility.kt:6]:
```suggestion
@Compatibility([Package("com.google.android.youtube", arrayOf("18.03.36"))])
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import app.revanced.patcher.extensions.replaceInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.annotation.HideSubtitlesPopupPatchCompatibility
+import app.revanced.patches.youtube.layout.hidesubtitlespopup.fingerprint.HideSubtitlesPopupFingerprint
+
+@Patch
+@Name("hide-subtitles-popup")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/hidesubtitlespopup/patch/HideSubtitlesPopupPatch.kt:17]: Please reflect the changes to the class and package name as well.
```suggestion
@Name("hide-subtitles-toggle-toast")
```

**Comment by MtndaleRedneck** [comment]
@oSumAtrIX why was this never instituted? Can it be?

**Comment by LisoUseInAIKyrios** [comment]
It's a simple patch any anyone can update and finish it.  All it needs is a preference menu item and an update to some of the 2 year old code that has since changed.

**Comment by Aunali321** [comment]
https://github.com/Aunali321/ReVancedExperiments/blob/main/patches/src/main/kotlin/li/auna/patches/youtube/layout/hidesubtitletoast/HideSubtitleToastPatch.kt

Here's the patch. Works fine. Patched last week.

**Pull request closed without merging**